backend/app/matrix/background_validator.py

```diff
@@ -0,0 +1,242 @@
+
+                if resp and is_team_allowed:
+                    return True
+
+            # Sponsorship tier conditions
+            sponsoring_tier = await gh_api.org_sponsored_at_tier(org_name)
+            if sponsoring_tier and org_conditions.sponsorship_tiers[sponsoring_tier]:
+                return True
+
+        # Organisation conditions
+        for user_name, user_conditions in room_github_conditions.users.items():
+
+            # Repository conditions
+            for repo_name, repo_conditions in user_conditions.repos.items():
+                resp = await gh_api.repo_permissions(user_name, repo_name, gh_username)
+
+                if resp == "admin" and repo_conditions.admin:
+                    return True
+                elif resp == "write" and repo_conditions.write:
+                    return True
+                elif resp == "read" and repo_conditions.read:
+                    return True
+
+            # Sponsorship tier conditions
+            sponsoring_tier = await gh_api.user_sponsored_at_tier(user_name)
+            if sponsoring_tier and user_conditions.sponsorship_tiers[sponsoring_tier]:
+                return True
+
+        return False
+
+    async def decide_next_state_of_user(
+        self, current_state: str, is_permitted: bool, disable_room_kick: str
+    ) -> str:
+        """
+        Returns the next state of the user (str).
+        """
+
+        if is_permitted:
+            if current_state == "join":
+                return current_state
+            else:
+                return "invite"
+        elif disable_room_kick:
+            return current_state
+        else:
+            return "leave"
+
+    async def send_next_state_event(self, room_id: str, user_id: str, next_state: str):
+        """
+        Sends a state event which sets the room membership for a user.
+        """
+
+        if next_state == "invite":
+            resp = await self.client.room_invite(room_id, user_id)
+
+            if isinstance(resp, RoomInviteError):
+                print(f"Error: {resp}")
+
+        elif next_state == "leave":
+            resp = await self.client.room_kick(room_id, user_id)
+
+            if isinstance(resp, RoomKickError):
+                print(f"Error: {resp}")
+
+    async def get_users_not_removed_by_bot(self, room_id: str) -> Set[str]:
+        """
+        Method that returns the set of users whose 'leave' state is not a result of the bot's actions.
+        """
+
+        ignore_members = set()
+
+        resp = await self.client.get_room_members(room_id)
+        if isinstance(resp, ErrorResponse):
+            print(f"Error: {resp}")
+
+        for room_member in resp.chunk:
+            # Bot is not the event's sender.
+            if room_member.sender != self.client.user_id:
+                ignore_members.add(room_member.state_key)
+
+        return ignore_members
+
+
+background_validator = BackgroundValidater()
```

Comment:
Why do we need this global? Can't we instantiate this when we start the background task? or if not put it in app state?